R/cohort_ref_class.R

```diff
@@ -0,0 +1,42 @@
+#' Cohort ref class
+#'
+#' This is a class to hold a list of pkg_refs along with extra
+#' information about the environment the packages are to be used/installed in.
+#'
+#' @export
+#' @family cohort_ref
+#'
+cohort_ref <- function(x, library, ...){
+  if (missing(x)) return(structure(list(pkg_ref_list = list(logical(0L)),
+                              library = logical(0L)), class = "cohort_ref"))
+  new_cohort_ref(x, library, ...)
+}
+
+new_cohort_ref <- function(x, library, ...){
+  dots <- list(...)
+  if (length(dots) && is.null(names(dots)) || any(names(dots) == ""))
+    stop("cohort_ref ellipses arguments must be named")
+  if(is.character(x) | all(sapply(x, class) != "pkg_ref")){
```

Comment:
Sorry, this is really offtopic, but seeing two different code styles for the `if` conditions (e.g. space VS no space after `if`, and requiring explicit curly braces even for atomic expressions VS skipping the optional curly brace in that case), I wonder if there are any plans to introduce `lintr` in the PR checks and enforce a common style?

Reply:
It wasn't on our roadmap, but I'm not opposed to adding it in. I generally find `lintr` to be a bit too opinionated and lacks a lot of the core functionality that makes other language linters/formatters useful (indentation, function header/call arg formatting, line-width aware linting).

Some of the lintrs that end up being particularly problematic are function name length (since it doesn't account for S3 class names used in dispatch) and camelCase/snake_case linters that throw lints for a lot of imported base functions.

I'd be fine adding in a selective set of linters for less-opinionated things like whitespace and line length.

Reply:
Yeah, I agree that `lintr` is far from perfect, and sometimes really annoying -- but I've found it very useful in collaborative projects to keep the code style/whitespace etc consistent and minimize the related (sometimes very opinionated) discussions once it's enforced.

E.g. setting custom line and object length, disabling some others (e.g. I'm a single-quote person) etc.

And still heavy use of `# nolint` for false positives, but at least that's explicit.